### PR TITLE
[CSLD-37] Fix and improve GetHeaders documentation

### PR DESCRIPTION
**Pull request by ChShersh**

### CODE REVIEW

File: _docs/technical/protocols/2017-01-13-binary.md [hunk 1]
```diff
@@ -1169,6 +1148,118 @@ data ConsensusData (GenesisBlockchain ssc) = GenesisConsensusData
...
+
+[//]: TODO: Describe Block
+
+## Block Exchange Messages
+
+### HeaderHash
+
+~~~ haskell
+-- | 'Hash' of block header. This should be @Hash (BlockHeader ssc)@
+-- but we don't want to have @ssc@ in 'HeaderHash' type.
```
Comment (ChShersh) [inline _docs/technical/protocols/2017-01-13-binary.md]: @rockbmb `ssc` parameter was removed here under this issue: 
https://issues.serokell.io/issue/CSL-604
I don't think reader of documentation interested in reasons why we are not using `HeaderHash` there. But I can fix comment to:

```haskell
-- | 'Hash' of block header. This should be @Hash (BlockHeader ssc)@
-- but there is no @ssc@ in HeaderHash type.
```

Comment (rockbmb) [inline _docs/technical/protocols/2017-01-13-binary.md]: @ChShersh If you do this today, I'll merge the PR myself afterwards.

Comment (ChShersh) [inline _docs/technical/protocols/2017-01-13-binary.md]: @rockbmb Done.

File: _docs/technical/protocols/2017-01-01-application-level.md [hunk 2]
```diff
@@ -110,14 +110,18 @@ module for the examples of messages that are using `Inv/Req/Data`.
...
-| GetBlocks | Oldest header hash; Newest hash | As opposed to GetHeaders, both hashes have to be present |
-| BlockHeaders | Non-empty collection of block headers, newest first | Polymorphic in scc |
-| Block | A single block | Polymorphic in scc |
+This table explains [Pos.Block.Network.Types](https://github.com/input-output-hk/cardano-sl/blob/3b657302dede832b908f7ba792a164c83b362712/src/Pos/Block/Network/Types.hs) module.
+
+| Message type | Payload                                                           | Commentaries                                             |
+|--------------+-------------------------------------------------------------------+----------------------------------------------------------|
+| GetHeaders   | Header hash checkpoints; Optional newest hash we're interested in | Expect newest header first                               |
+| GetBlocks    | Oldest header hash; Newest hash                                   | As opposed to GetHeaders, both hashes have to be present |
+| BlockHeaders | Non-empty collection of block headers, newest first               | Polymorphic in scc                                       |
```
Comment (rockbmb) [inline _docs/technical/protocols/2017-01-01-application-level.md]: I only now noticed this line and the next, but `scc` -> `ssc`.

Comment (ChShersh) [inline _docs/technical/protocols/2017-01-01-application-level.md]: Fixed such typos in docs.

File: _docs/technical/protocols/2017-01-01-application-level.md [hunk 3]
```diff
@@ -110,14 +110,18 @@ module for the examples of messages that are using `Inv/Req/Data`.
...
-| GetHeaders | Oldest header hash we're interested in; Newest hash we're interested in, or blank | Expect newest header first |
-| GetBlocks | Oldest header hash; Newest hash | As opposed to GetHeaders, both hashes have to be present |
-| BlockHeaders | Non-empty collection of block headers, newest first | Polymorphic in scc |
-| Block | A single block | Polymorphic in scc |
+This table explains [Pos.Block.Network.Types](https://github.com/input-output-hk/cardano-sl/blob/3b657302dede832b908f7ba792a164c83b362712/src/Pos/Block/Network/Types.hs) module.
+
+| Message type | Payload                                                           | Commentaries                                             |
+|--------------+-------------------------------------------------------------------+----------------------------------------------------------|
+| GetHeaders   | Header hash checkpoints; Optional newest hash we're interested in | Expect newest header first                               |
+| GetBlocks    | Oldest header hash; Newest hash                                   | As opposed to GetHeaders, both hashes have to be present |
```
Comment (gromakovsky) [inline _docs/technical/protocols/2017-01-01-application-level.md]: I don't like this _As opposed to GetHeaders_. These messages have different arguments and different semantics. I would understand it if `GetHeaders` had two optional hashes (or one optional and one mandatory). But `GetHeaders` has a list of checkpoints and optional newest hash, while `GetBlocks` has two hashes which specify range.

File: _docs/technical/protocols/2017-01-13-binary.md [hunk 4]
```diff
@@ -1169,6 +1148,118 @@ data ConsensusData (GenesisBlockchain ssc) = GenesisConsensusData
...
+|------------------+-------------+--------------------+----------------------------------------------|
+| 1-9              | UVarInt Int | n                  | Number of checkpoints                        |
+| n * size(Hash)   | Hash[n]     |                    | List of length `n` with hashes               |
+| 1                | Word8       | tag = 0x00 or 0x01 | Tag for optional `to` hash                   |
+| tag * size(Hash) | Hash        |                    | If tag is not `0x00` than hash of `to` block |
+
+### GetBlocks
+
+~~~ haskell
+-- | 'GetHeaders' message (see protocol specification).
```
Comment (gromakovsky) [inline _docs/technical/protocols/2017-01-13-binary.md]: Copy-paste (`GetBlocks`)

File: _docs/technical/protocols/2017-01-13-binary.md [hunk 5]
```diff
@@ -1169,6 +1148,118 @@ data ConsensusData (GenesisBlockchain ssc) = GenesisConsensusData
...
+    , mghTo   :: !(Maybe HeaderHash)
+    } deriving (Generic, Show, Eq)
+~~~
+
+| Field size       | Type        | Value              | Description                                  |
+|------------------+-------------+--------------------+----------------------------------------------|
+| 1-9              | UVarInt Int | n                  | Number of checkpoints                        |
+| n * size(Hash)   | Hash[n]     |                    | List of length `n` with hashes               |
+| 1                | Word8       | tag = 0x00 or 0x01 | Tag for optional `to` hash                   |
+| tag * size(Hash) | Hash        |                    | If tag is not `0x00` than hash of `to` block |
```
Comment (gromakovsky) [inline _docs/technical/protocols/2017-01-13-binary.md]: then

File: _docs/technical/protocols/2017-01-13-binary.md (hunk 4 above)
Comment (ChShersh) [inline _docs/technical/protocols/2017-01-13-binary.md]: Thanks! I fixes this in code but forgot to fix in docs

**Pull request merged**